### PR TITLE
feat(anta.tests): Add IS-IS tests collection

### LINKED ISSUE #636: Features: Implement ISIS tests for a standard ISIS-SR environment
## ISIS tests

- [x] Check that all ISIS neighbors are UP
- [x] Number of peers per interface
- [x] List of passive/non-passive interfaces

## Segement Routing

TBD

### PR DESCRIPTION
**Pull request by titom73**
# Description

Implement new set of tests to cover IS-IS and Segment Routing

## IS-IS tests

- [x] Check that all ISIS neighbors are UP
- [x] Number of peers per interface
- [x] List of passive/non-passive interfaces

Related to #636

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have run pre-commit for code linting and typing (`pre-commit run`)
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes (`tox -e testenv`)

### CODE REVIEW

File: anta/tests/routing/isis.py [hunk 1]
```diff
@@ -0,0 +1,307 @@
...
+                )
+
+
+class VerifyISISInterfaceMode(AntaTest):
+    """Verifies ISIS Interface are running in correct mode.
+
+    Expected Results
+    ----------------
+    * Success: The test will pass if all listed interfaces are running in correct mode.
+    * Failure: The test will fail if one of the listed interfaces is not running in correct mode.
```
Comment (gmuloc) [inline anta/tests/routing/isis.py:228]:
```suggestion
    * Failure: The test will fail if any of the listed interfaces is not running in correct mode.
```

File: anta/tests/routing/isis.py [hunk 2]
```diff
@@ -0,0 +1,307 @@
...
+        interfaces: list[InterfaceState]
+        """list of interfaces with their information."""
+
+        class InterfaceState(BaseModel):
+            """Input model for the VerifyISISNeighborCount test."""
+
+            name: Interface
+            """Interface name to check."""
+            level: Literal[1, 2] = 2
+            """xxx."""
```
[the remark this answers is not in the document]
Comment (titom73) [reply anta/tests/routing/isis.py:269]: Fixed

File: anta/tests/routing/isis.py [hunk 3]
```diff
@@ -0,0 +1,307 @@
...
+            # Check for broadcast
+            elif interface.mode == "broadcast":
+                if interface_data is None or get_value(dictionary=interface_data, key="interfaceType", default="unset") != "broadcast":
+                    self.result.is_failure(f"Interface {interface.name} in vrf {interface.vrf} is not running in broadcast mode")
+
+            # Check for passive
+            elif interface.mode == "passive":
+                json_path = f"intfLevels.{interface.level}.passive"
+                if interface_data is None or get_value(dictionary=interface_data, key=json_path, default=False) is False:
+                    self.result.is_failure(f"Interface {interface.name} in vrf {interface.vrf} is not running in passive mode")
```
[the remark this answers is not in the document]
Comment (titom73) [reply anta/tests/routing/isis.py]: Function updated with your approach

[the remark this answers is not in the document]
Comment (titom73) [reply anta/tests/routing/isis.py]: `p2p` has been changed to `point-to-point`

File: anta/tests/routing/isis.py [hunk 4]
```diff
@@ -0,0 +1,307 @@
...
+            """Number of IS-IS neighbors."""
+
+    @AntaTest.anta_test
+    def test(self) -> None:
+        """Main test function for VerifyISISNeighborCount."""
+        command_output = self.instance_commands[0].json_output
+        self.result.is_success()
+        isis_neighbor_count = _get_isis_neighbors_count(command_output)
+        if len(isis_neighbor_count) == 0:
+            self.result.is_skipped("No IS-IS neighbor detected")
```
[the remark this answers is not in the document]
Comment (titom73) [reply anta/tests/routing/isis.py:213]: I would say it depends how you build your inventory:
- All tests everywhere and let tests skip if feature is not configured
- Ensure you configure tests only on device where feature should be active

**Pull request merged**